themes/default/content/blog/converting-full-terraform-programs-to-pulumi/index.md

```diff
@@ -0,0 +1,184 @@
+---
+title: "Converting Full Terraform Programs to Pulumi"
+date: 2023-06-12
+meta_desc: Learn how to convert whole Terraform programs to Pulumi using the new Terraform converter
+
+
+meta_image: meta.png
+
+authors:
+    - justin-vanpatten
+
+tags:
+    - migration
+---
+
+Today, we're excited to announce support for converting whole Terraform projects to Pulumi via the `pulumi convert` command in the Pulumi CLI. The new Terraform converter includes support for Terraform modules, all core features of Terraform 1.4, and the majority of Terraform built-in functions. Let's dig in to learn more about the new converter and how to use it.
+
+<!--more-->
+
+Historically, we have offered a separate [tf2pulumi](https://github.com/pulumi/tf2pulumi) tool to convert snippets of Terraform to Pulumi. Today we're announcing a new converter that supports converting the vast majority of complete Terraform projects, including modules. There's no longer a separate tool to download -- you can run the new converter directly from the Pulumi CLI with the `pulumi convert --from terraform` command to convert Terraform to Pulumi TypeScript, Python, Go, or C#.
```

Comment:
In the beginning of the blog: "Today, we're excited to announce support" then in the second paragraph "Today we're announcing a new converter". I suggest rewriting the paragraph without repeating:

> Historically, we have offered a separate [tf2pulumi](https://github.com/pulumi/tf2pulumi) tool to convert snippets of Terraform code to Pulumi. However, we realized that having a separate CLI for conversion made this feature less accessible for the majority of Pulumi users who don't want to download `tf2pulumi` and keep it up to date. Instead, you can run the new converter directly from the Pulumi CLI with the `pulumi convert --from terraform` command to convert Terraform to Pulumi TypeScript, Python, Go, or C#. 
> 
> This also means that from now on, it will be much easier for us to deliver updates and improvements to the converter which users will get when they update their Pulumi CLI.

Reply:
Actually convert fixes are now decoupled from cli fixes to some degree. We should make this sound like we can do fixes even easier.

Reply:
Tweaked the text here so it's not repeating.